### PR TITLE
refactor: replace RecursiveMutex `cs_vProcessMsg` with Mutex (and rename)

### PR DESCRIPTION
**Pull request by theStack**
This PR is related to #19303 and gets rid of the RecursiveMutex cs_vProcessMsg. Both of the critical sections only directly access the guarded elements, i.e. it is not possible that within one section another one is called, and we can use a regular Mutex:
https://github.com/bitcoin/bitcoin/blob/e3ce019667fba2ec50a59814a26566fb67fa9125/src/net.cpp#L1597-L1602
https://github.com/bitcoin/bitcoin/blob/e3ce019667fba2ec50a59814a26566fb67fa9125/src/net_processing.cpp#L4142-L4150
Also, it is renamed to adapt to the (unwritten) naming convention to use the `_mutex` suffix rather than the `cs_` prefix.

### CODE REVIEW

**Review by shaavan — COMMENTED** [review]
Concept ACK 

I was thinking, should we use `AssertLockNotHeld(m_process_msgs_mutex)` before locking the mutex and add thread safety annotation `LOCKS_EXCLUDED(m_process_msgs_mutex)`, with the function declaration of where this mutex is used?

Doing so will prevent any risk of error due to simultaneous locking of this mutex.

**Review by w0xlt — APPROVED** [review]
crACK 9a355e1

I agree with @shaavan .

It might be nice to add `AssertLockNotHeld(mutex)` macros combined with `LOCKS_EXCLUDED(mutex)` thread safety annotations. This avoids any risk of recursive locking.

**Comment by theStack** [comment]
@shaavan @w0xlt: Thanks for your feedback, I agree that adding thread safety assertions and annotations is a good idea. Added a commit doing that -- note though that the `LOCKS_EXCLUDED` annotation is only possible for one method (`PeerManagerImpl::ProcessMessages`), as for the other (`CConnman::SocketHandlerConnected`) the elements that cointain the loop are in a list rather than a direct parameter that can be accessed.

File: src/test/util/net.cpp [hunk 1]
```diff
@@ -22,8 +22,8 @@ void ConnmanTestMsg::NodeReceiveMsgBytes(CNode& node, Span<const uint8_t> msg_by
             nSizeAdded += it->m_raw_message_size;
         }
         {
-            LOCK(node.cs_vProcessMsg);
-            node.vProcessMsg.splice(node.vProcessMsg.end(), node.vRecvMsg, node.vRecvMsg.begin(), it);
+            LOCK(node.m_process_msgs_mutex);
```
Comment (shaavan) [inline src/test/util/net.cpp:25]: Can we add `AssertLockNotHeld` above this line?

```suggestion
            AssertLockNotHeld(node.m_process_msgs_mutex);
            LOCK(node.m_process_msgs_mutex);
```

**Review by shaavan — COMMENTED** [review]
> as for the other (CConnman::SocketHandlerConnected) the elements that cointain the loop are in a list rather than a direct parameter that can be accessed.

Sounds interesting. Let me take another look at this before I can be 100% sure. In the meantime, I had one observation that I would like to discuss.

**Review by hebasto — COMMENTED** [review]
Concept ACK.

Following best practices, mutexes should synchronize access to internal class data to prevent violation of class invariants.

The current implementation has a significant design drawback -- external synchronizing, i.e., acquiring `CNode::m_process_msgs_mutex` from outside of the `CNode` instance.

Therefore, I'm suggesting to consider to encapsulate access to the private members `m_process_msgs` and `nProcessQueueSize` with internal synchronizing using `m_process_msgs_mutex`.

One new `CNode` member function could implement https://github.com/bitcoin/bitcoin/blob/e3de7cb9039770e0fd5b8bb8a5cba35c87ae8f00/src/net.cpp#L1592-L1604

Another one https://github.com/bitcoin/bitcoin/blob/e3de7cb9039770e0fd5b8bb8a5cba35c87ae8f00/src/net_processing.cpp#L4165-L4173

**Comment by achow101** [comment]
Are you still working on this?

**Comment by achow101** [comment]
Closing this as it has not had any activity in a while. If you are interested in continuing work on this, please leave a comment so that it can be reopened.

**Pull request closed without merging**